### PR TITLE
ssbc: Step-wise caching

### PR DESCRIPTION
**Pull request by eseliger**
This PR implements the main-repo changes required for step-wise caching.

Requires https://github.com/sourcegraph/src-cli/pull/658 and https://github.com/sourcegraph/src-cli/pull/659

Closes https://github.com/sourcegraph/sourcegraph/issues/26932
Closes https://github.com/sourcegraph/sourcegraph/issues/26933

### CODE REVIEW

File: enterprise/cmd/frontend/internal/executorqueue/queues/batches/transform.go [hunk 1]
```diff
@@ -9,51 +9,29 @@ import (
...
 	apiclient "github.com/sourcegraph/sourcegraph/enterprise/internal/executor"
 	"github.com/sourcegraph/sourcegraph/internal/actor"
+	"github.com/sourcegraph/sourcegraph/internal/api"
 	"github.com/sourcegraph/sourcegraph/internal/conf"
 	"github.com/sourcegraph/sourcegraph/internal/database"
 	batcheslib "github.com/sourcegraph/sourcegraph/lib/batches"
+	"github.com/sourcegraph/sourcegraph/lib/batches/execution/cache"
 )
 
-const (
```
Comment (eseliger) [inline enterprise/cmd/frontend/internal/executorqueue/queues/batches/transform.go:21]: Moved to bottom of file

**Comment by eseliger** [comment]
Note to self: Bump src-cli before merging.

**Review by mrnugget — APPROVED** [review]
Nice! Well done!

File: enterprise/cmd/frontend/internal/batches/resolvers/batch_spec_workspace.go [hunk 2]
```diff
@@ -99,6 +118,28 @@ func (r *batchSpecWorkspaceResolver) computeStepResolvers(ctx context.Context) (
...
+	if err != nil {
+		return nil, err
+	}
+
+	spec, err := r.store.GetBatchSpec(ctx, store.GetBatchSpecOpts{ID: r.workspace.BatchSpecID})
+	if err != nil {
+		return nil, err
+	}
+
+	taskKey := service.CacheKeyForWorkspace(spec, &service.RepoWorkspace{
```
[the remark this answers is not in the document]
Comment (eseliger) [reply enterprise/cmd/frontend/internal/batches/resolvers/batch_spec_workspace.go:131]: Good idea! I actually started doing that in a follow up where I move things around to prepare for the bonus target so I'll include that in there :)

**Pull request merged**